ratecounter.go

```diff
@@ -76,6 +76,18 @@ func (r *RateCounter) Rate() int64 {
 	return r.counter.Value()
 }
 
+func (r *RateCounter) MaxTickRate() int64 {
+	max := int64(0)
+
+	for _, p := range r.partials {
+		if max < p.Value() {
```

Comment:
Probably want to capture this the first time you read it:
```
if val := p.Value(); max < val {
  max = val
}
```
so it can't change in the meantime.